### ISSUE TITLE
i3: Multiple missing xcb utils, resolvable with pacman, then 'no such file'

### ISSUE THREAD

**Issue by OJFord** [Linuxbrew/homebrew-core#2738]
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [N/A] Attempt to replicate the problem on macOS with Homebrew if possible. If you can replicate the problem on macOS with Homebrew, please report the issue to [Homebrew/core](https://github.com/Homebrew/homebrew-core). If you are unsure, please report the issue to Linuxbrew.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [N/A] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

https://gist.github.com/91e49b32fda573f3cbd3109320947d97

`i3` fails first in installing its dependency, `startup-notification`, which complains of missing `xcb-aux`. This is resolved by installing `xcb-util` from pacman - presumably a newer version than that obtained via the brew dependency.

With `startup-notification` installed, installation falls down in `i3` itself: first with no `xcb-cursor` (the package `xcb-util-cursor` is available via pacman, but not brew) and then no `xcb-keysyms`. Again, `xcb-util-keysyms` from brew seems no good, but fine from pacman. Then, `xcb-icccm` is missing, resolved by installing `xcb-util-wm` from pacman (not brew).

In summary, some dependencies seemingly need updating:
- `xcb-util`
- `xcb-util-keysyms`
- `xcb-util-wm`

and one is missing altogether; which should be a dependency of `i3`:
- `xcb-util-cursor`

After all that:

> Error:  No such file or directory @ dir_chdir - x86_64-apple-darwin4.11.2-1-ARCH

and I'm stumped.

**Comment by sjackman** [comment]
Change this line:
https://github.com/Linuxbrew/homebrew-core/blob/master/Formula/i3.rb#L48
```ruby
cd "x86_64-apple-darwin#{`uname -r`.chomp}" do
```
to I'm guessing something like
```ruby
cd `./config.guess` do
```
A pull request would be welcome if you're able to fix it.

**Comment by sjackman** [comment]
@maxim-belkin Any thoughts on the `xcb` issues above?

**Comment by maxim-belkin** [comment]
@sjackman `brew tap linuxbrew/xorg` ?

**Comment by maxim-belkin** [comment]
@OJFord `xcb-*` dependencies are provided through `linuxbrew/xorg` tap. 
Could you please do the following:
```
brew linkage i3
```
If you are adventurous, try removing all `xcb*` packages that you installed with pacman, then `brew tap linuxbrew/xorg` and then install all `*xcb*` packages from that tap, and then do `brew install i3`.
